Tasks/IISWebAppDeploymentOnMachineGroup/deployiiswebapp.ts

```diff
@@ -32,6 +40,42 @@ async function run()
 		webDeployPkg = availableWebPackages[0];
 
 		var isFolderBasedDeployment = await utility.isInputPkgIsFolder(webDeployPkg);
+
+        if(JSONFiles.length != 0 || xmlTransformation || xmlVariableSubstitution) {
+
+            var folderPath = path.join(tl.getVariable('System.DefaultWorkingDirectory'), 'temp_web_package_folder');
+            if(isFolderBasedDeployment) {
+                tl.cp(path.join(webDeployPkg, '/*'), folderPath, '-rf', false);
+            }
+            else {
+                await zipUtility.unzip(webDeployPkg, folderPath);
+            }
+
+            if(xmlTransformation) {
+                var environmentName = tl.getVariable('Release.EnvironmentName');
+                if(tl.osType().match(/^Win/)) {
+                    var transformConfigs = ["Release.config"];
+                    if(environmentName) {
+                        transformConfigs.push(environmentName + ".config");
+                    }
+                    xdtTransformationUtility.basicXdtTransformation(path.join(folderPath,'**', '*.config'), transformConfigs);  
+                    tl._writeLine("XDT Transformations applied successfully");
+                } else {
+                    throw new Error(tl.loc("CannotPerformXdtTransformationOnNonWindowsPlatform"));
+                }
+            }
+
+            if(xmlVariableSubstitution) {
+                await xmlSubstitutionUtility.substituteAppSettingsVariables(folderPath);
+            }
+
+            if(JSONFiles.length != 0) {
+                jsonSubstitutionUtility.jsonVariableSubstitution(folderPath, JSONFiles);
+            }
+
+            webDeployPkg = (isFolderBasedDeployment) ? folderPath : await zipUtility.archiveFolder(folderPath, tl.getVariable('System.DefaultWorkingDirectory'), 'temp_web_package.zip')
```

Comment:
The idea of using common modules is not happening if duplicate. This task should not have any thing more than Task.json and corresponding read input changes for new feature additions. We have to make the code that way.